### PR TITLE
BUGFIX: SD-502 update error status for export

### PR DESCRIPTION
**Pull request by lvkhanh**
# PR checklist

- [x] Change was tested using dev-release in Analytical Designer and Dashboards (if applicable).
- [x] Change is described in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [x] Migration guide (for major update) is written to [CHANGELOG.md](../blob/master/CHANGELOG.md).

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/294/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/350/ : SUCCESS in 1m 47s

File: test/report.test.ts [hunk 1]
```diff
@@ -88,9 +88,13 @@ describe('report', () => {
...
                 return mockedReportModule()
                     .exportResult(projectId, executionResult, exportConfig, { pollStep: 1 })
-                    .then(null, (error: Error) => expect(error.message).toBe(
-                        'You cannot export this insight because it contains restricted data.'
-                    ));
+                    .then(null, (error: ApiResponseError) => {
+                        expect(error.response.status).toBe(400);
+                        expect(error.response.statusText).toBe('Bad Request');
+                        expect(error.message).toBe(
+                            'You cannot export this insight because it contains restricted data.'
```
Comment (naibinh) [inline test/report.test.ts]: it should be `.toBe(ERROR_RESTRICTED_MESSAGE)`

File: src/report.ts [hunk 2]
```diff
@@ -77,7 +77,7 @@ export class ReportModule {
             error.response.status === 400 &&
             error.responseBody.indexOf(ERROR_RESTRICTED_CODE) !== -1
         ) {
-            return Promise.reject(new Error(ERROR_RESTRICTED_MESSAGE));
+            error.message = ERROR_RESTRICTED_MESSAGE;
```
Comment (naibinh) [inline src/report.ts]: I'm not really keen on modifying param `error` in this case.
I think current way is ok, or we have to clone to new `error` object.

File: test/report.test.ts (hunk 1 above)
Comment (lvkhanh) [reply test/report.test.ts]: updated :)

File: src/report.ts (hunk 2 above)
Comment (lvkhanh) [reply src/report.ts]: update :)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/300/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/357/ : SUCCESS in 2m 24s

**Comment by naibinh** [comment]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by lvkhanh** [comment]
I squashed commit

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-merge-compliance-check/83/ : SUCCESS in 26s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/365/ : SUCCESS in 6m 09s

**Pull request merged**